### ISSUE TITLE
Triggering actions in 3rd party applications via Obj-C

### ISSUE THREAD

**Issue by latenitefilms** [Hammerspoon/hammerspoon#1510]
Sorry, this might not be specifically related to Hammerspoon, but I figure if anyone knows the answer it will be @cmsj or @asmagill.

Is it possible to "trigger" another applications internal "functions" using Objective-C. Let's take TextEdit for example. Let's say I know that in TextEdit (thanks to some detective work), that when you press `Text Edit > About TextEdit`, it triggers an "action" called `orderFrontAboutBox`. Is it possible to call this "action" (for lack of knowledge to use the right terminology) from an Hammerspoon extension in Objective-C?

Basically... what I'm HOPING to do is that I could use Objective-C calls to do things, to avoid having to have Accessibility permissions and rely on GUI Scripting.

Any ideas? Possible?

**Comment by asmagill** [comment]
Basically no, not without some serious code injection and hijacking that would be brittle, unreliable, and likely to break with every update (app or OS).  That's why applications opt in to supporting things like AppleScript and Accessibility -- to allow outside applications limited ability to control them in ways the developer considers "safe".

It comes down to application security -- anything Hammerspoon could do with the best of intentions, someone else could do with less noble aspirations.  At various points in macOS history there have been plugins and bundle loaders (SIMBL comes to mind) that provided some ways to work around this, but they are brittle, usually require updates or rewrites with OS updates and have become more and more difficult to implement with app sandboxing, SIP, etc.  Last I checked SIMBL only worked up to 10.9 (and even then only partially) and hadn't been updated in years.

**Comment by latenitefilms** [comment]
@asmagill - Legend, thank you! Great information.
